src/components/Game/Game.js

```diff
@@ -1,35 +1,43 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 
 import classes from './Game.scss';
 import { Slots } from './Slots/Slots';
 import { Letters } from './Letters/Letters';
+import Result from './Result/Result';
 
 import { WORD_LENGTH } from '../../constants/constants';
 import { shuffleArray, generateID } from '../../helpers';
-import { getCorrectWords } from '../../service/service';
-
+import { getCorrectWordsAction } from '../../store/actions';
 
 class Game extends Component {
     state = {
-        correctWords: [],
         letters: [],
         lettersInSlots: [],
         gameWon: undefined,
     };
 
     componentDidMount() {
-        const correctWords = getCorrectWords();
-        const lettersObjects = correctWords[0].split('')
-            .map((letter) => ({
-                value: letter,
-                id: generateID(),
-                active: true,
-            }));
+        const { getCorrectWords } = this.props;
+        getCorrectWords();
+    }
 
-        this.setState({
-            correctWords,
-            letters: shuffleArray(lettersObjects),
-        });
+    componentDidUpdate(prevProps) {
+        const { correctWords } = this.props;
+
+        if (correctWords !== prevProps.correctWords) {
+            const lettersObjects = correctWords[0].split('')
+                .map((letter) => ({
+                    value: letter,
+                    id: generateID(),
+                    active: true,
+                }));
+            /* eslint-disable react/no-did-update-set-state */
+            this.setState({
+                letters: shuffleArray(lettersObjects),
+            });
```

Comment:
extracting lines 30-39 to a separate method `initLetters` would make code more readable and self-explanatory